library_api/router/router.go

```diff
@@ -10,7 +10,7 @@ import (
 )
 
 func Route() {
-	b := controller.Book{Book_repository: repository.NewBookRepository()}
+	b := controller.Book{Book_repository: repository.NewCsvRepository()}
```

Comment:
nice, you haven't needed to change the interface signature )) - well done
So now you understand how repository works related to
"Different implementations - same interface"